docs/cloud/secret-management.md

```diff
@@ -0,0 +1,92 @@
+# Secret management
+
+:::caution Under Construction
+
+This draft is for demo purposes with test setups.
+Secret management is under ongoing development.
+Be sure to take the [important configuration steps](https://kubernetes.io/docs/concepts/configuration/secret/) and follow the [good practices](https://kubernetes.io/docs/concepts/security/secrets-good-practices/) defined in the Kubernetes documentation to safely use secrets.
+
+:::
+
+:::caution Experimental feature
+
+Secret management is an experimental feature of Feldera. Setting up secrets
+currently involves a number of manual steps, which will be integrated
+into an automated/improved workflow in the production release.
+
+:::
+
+Within certain connector configurations, it is possible to refer to secrets rather than inputting them as plain text. This prevents them from being visible in the connector configurations. We currently make use of [Kubernetes secrets](https://kubernetes.io/docs/concepts/configuration/secret/) for management.
+
+
+## Feldera connector secrets
+
+Referring to secrets is currently enabled only for the Feldera Kafka input and output connector configuration.
+
+1. **Create kubernetes secret:** create a secret named `example` within the `feldera` namespace, which has a single key `example` with as value `something` (base64 encoded: `c29tZXRoaW5n`).
+   We can do this by creating a file `example-secret.yaml`:
+   ```
+   apiVersion: v1
+   kind: Secret
+   metadata:
+     name: example
+     namespace: feldera
+   type: Opaque
+   data:
+     example: c29tZXRoaW5n
+   ```
+
+   ... and subsequently apply it using: `kubectl apply -n feldera -f example-secret.yaml`
+
+2. **Use the secret:** now, when we start the pipeline with a connector configuration value of `${secret:example}`, it will be resolved using a mounted secret volume containing the secret we just defined.
+
+
+## Kafka Connect (Debezium) secrets
+
+When creating a Kafka Connect Debezium connector to for example MySQL, it is required to provide database credentials. We make use of the Kafka Connect [FileConfigProvider](https://docs.confluent.io/platform/current/connect/security.html#fileconfigprovider), which allows us to refer to secrets in a similar way as Feldera connector secrets (with a slightly different pattern).
+
+Because Kafka Connect is already running, it is not possible to mount new secrets without restarting. Instead, we patch an already existing mounted secret and await for it to be synchronized with the running instance.
+
+**The steps are as follows:**
+
+1. The secret used is `kafka-connect-databases-secret` in the `kafka-connect` namespace.
+   It is mounted at `/etc/secret-volume-kafka-connect-databases` on the Kafka Connect instance.
```

Comment:
The operator may not know what Kafka Connect is or why it exists in the install. I think we need a software bill of materials (BOM) to describe the k8s services (and pods) that come up with Feldera Cloud and the persistent volumes it uses and under those which OS mounts those use.

Reply:
This is temporary. We will likely manage Kafka Connect via the backend and then we'll fold this into a BOM then.